lib/awful/util.lua.in

```diff
@@ -46,14 +48,13 @@ function util.deprecate(see)
     io.stderr:write("\n" .. tb .. "\n")
 end
 
---- Strip alpha part of color.
+--- Check if a color is valid in Pango markup.
 -- @param color The color.
--- @return The color without alpha channel.
-function util.color_strip_alpha(color)
-    if color:len() == 9 then
-        color = color:sub(1, 7)
-    end
-    return color
+-- @return True if the color valid, else nil.
+function util.check_pango_color(color)
+    if not color then return nil end
+    local attr, _ = Pango.parse_markup("<span background=\"" .. color .. "\"></span>", -1, 0)
```

Comment:
I thought about that, but it looks to me like most code in awesome doesn't type check and just assumes that colors are either strings or nil, so that's what I accounted for.